src/WebJobs.Script/Description/CSharp/CSharpFunctionSignature.cs

```diff
@@ -53,10 +53,7 @@ public static CSharpFunctionSignature FromCompilation(Compilation compilation, I
                 throw new ArgumentException("The provided compilation does not have a syntax tree.", "compilation");
             }
 
-            SyntaxTree tree = compilation.SyntaxTrees.First();
-            SemanticModel model = compilation.GetSemanticModel(tree);
-
-            var methods = ((INamedTypeSymbol)model.GetEnclosingSymbol(0))
+            var methods = compilation.ScriptClass
```

Comment:
This change actually gives us some performance benefits as well, since it now replaces a call to create get the semantic model, which was showing to be an expensive one.